File: test_project/.vscode/SnipHub/snip/hello.php
<?php
// PHP Hello World
echo "Hello, World!\n";

// 带函数的 Hello World
function sayHello($name) {
    return "Hello, $name!";
}

echo sayHello("PHP") . "\n";

// 带类的 Hello World
class Greeter {
    private $name;
    
    public function __construct($name) {
        $this->name = $name;
    }
    
    public function greet() {
        return "Hello, {$this->name}!";
    }
}

// 创建实例并调用方法
$greeter = new Greeter("PHP OOP");
echo $greeter->greet() . "\n";

// 使用数组
$languages = ["PHP", "HTML", "JavaScript", "CSS"];
foreach ($languages as $lang) {
    echo "I know $lang!\n";
}
?>

File: test_project/.vscode/SnipHub/snip/hello.rb
# Ruby Hello World
puts "Hello, World!"

# 带方法的 Hello World
def say_hello(name)
  "Hello, #{name}!"
end

puts say_hello("Ruby")

# 带类的 Hello World
class Greeter
  def initialize(name)
    @name = name
  end
  
  def greet
    "Hello, #{@name}!"
  end
end

# 创建实例并调用方法
greeter = Greeter.new("Ruby OOP")
puts greeter.greet

# 使用代码块
5.times do |i|
  puts "#{i+1}. Hello from Ruby!"
end
